lib/genome/importers/human_protein_atlas/human_protein_atlas.rb

```diff
@@ -0,0 +1,70 @@
+require 'genome/online_updater'
+
+module Genome; module Importers; module HumanProteinAtlas;
+  class HumanProteinAtlas < Genome::OnlineUpdater
+    attr_reader :file_path, :source
+
+    def initialize(file_path)
+      @file_path = file_path
+    end
+
+    def import
+      remove_existing_source
+      create_new_source
+      create_gene_claims
+    end
+
+    private
+    def remove_existing_source
+      Utils::Database.delete_source('HumanProteinAtlas')
+    end
+
+    def create_new_source
+      @source ||= DataModel::Source.create(
+        {
+            base_url: 'https://www.proteinatlas.org/search/protein_class%3APotential+drug+targets',
+            site_url: 'https://www.proteinatlas.org/',
+            citation: 'Uhlén M, Fagerberg L, Hallström BM, et al. Proteomics. Tissue-based map of the human proteome. Science. 2015;347(6220):1260419. doi:10.1126/science.1260419. PMID: 25613900',
+            source_db_version:  '19.3',
+            source_type_id: DataModel::SourceType.POTENTIALLY_DRUGGABLE,
+            source_db_name: 'HumanProteinAtlas',
+            full_name: 'The Human Protein Atlas',
+            license: 'Creative Commons Attribution-ShareAlike 3.0 International License',
+            license_link: 'https://www.proteinatlas.org/about/licence',
+        }
+      )
+    end
+
+    def create_gene_claims
+      CSV.foreach(file_path, :headers => true, :col_sep => "\t") do |row|
+        gene_claim = create_gene_claim(row["Gene"], 'Gene Symbol')
+        create_gene_claim_alias(gene_claim, row["Ensembl"], 'Ensembl Gene ID')
+        unless row['Gene synonym'].nil?
+          row['Gene synonym'].split(', ').each do |s|
+            create_gene_claim_alias(gene_claim, s, 'Human Protein Atlas Gene Synonym')
+          end
+        end
+        create_gene_claim_alias(gene_claim, row['Gene description'], 'Human Protein Atlas Gene Description')
+        create_gene_claim_alias(gene_claim, row['Uniprot'], 'UniProt ID')
+
+        row['Protein class'].split(', ').each do |c|
+          if categories.has_key? c
+            gene_category = DataModel::GeneClaimCategory.find_by(name: categories[c])
+            gene_claim.gene_claim_categories << gene_category unless gene_claim.gene_claim_categories.include? gene_category
+          end
+        end
+      end
+    end
+
+    def categories
+      {
```

Comment:
I don't know how many rows there are in this dataset, and either way the impact is likely minimal but this method will create and retunr a new dictionary every time its called, which appears to be 2x per protein class per row which could add up. Memoizing it with 

```
def categories
  @categories ||= { 
     ....dictionary contents
}
end
```

Will only create the hash the first call and then just return a reference to that one each time its called. If it runs satisfactorily fast currently, you may not have to care though,

Reply:
Done.